modules/ServerAPI/src/server/routes/ExperimentServiceRoutes.coffee

```diff
@@ -785,41 +785,171 @@ exports.createAndUpdateExptExptItxs = (req, resp) ->
+			if acls.getRead()
+				# Check to see if the 
+				acls.setWrite(exports.canEditExperiment(user, experiment, projectCode, scientist))
+				acls.setDelete(exports.canDeleteExperiment(user, projectCode, scientist))
+
+	console.log "Experiment #{experiment.codeName} has acls: #{JSON.stringify(acls)} for user #{user.username}"
+	return acls
+			
+exports.canEditExperiment = (user, experiment, project, scientist) ->
+	# If editing roles are not defined, then allow editing
+	# If editing roles are define then
+	#    If the role is "entityScientist" then the user.username just needs to be the same as the scientist
+	#    If the role is "projectAdmin" then the user needs to have the project admin role for the project
+	# Order seems to really matter here, but this is what was done in the old code so I'm leaving it as is.
+	if experiment.lsKind is 'study'
+		editingRoles = null
+		if config.all.entity?.study?.editingRoles?
+			editingRoles = config.all.entity.study.editingRoles
+	else if config.all.entity?.editingRoles?
+		editingRoles = config.all.entity.editingRoles
+	if editingRoles?
+		rolesToTest = []
+		for role in editingRoles.split(",")
+			role = role.trim()
+			if role is 'entityScientist'
+				if (user.username == scientist)
+					return true
+			else if role is 'projectAdmin'
+				projectAdminRole =
+					lsType: "Project"
+					lsKind: project
+					roleName: "Administrator"
+				if testUserHasRoleTypeKindName(user, [projectAdminRole])
+					return true
 			else
-				console.log "Failed: got error in bulk get of experiments"
-				callback "Bulk get experiments saveFailed: " + JSON.stringify error
-		)
+				rolesToTest.push role
+		if rolesToTest.length is 0
+			return false
+		unless testUserHasRole user, rolesToTest
+			return false
+	return true
+
+exports.canDeleteExperiment = (user, project, scientist) ->
+	# If deleting roles are not defined, then allow delete
+	# If deleting roles are defined then 
+	#   If the role is "entityScientist" then the user.username just needs to be the same as the scientist
+	#   If the role is "projectAdmin" then the user needs to have the project admin role for the project
+	# Order seems to really matter here, but this is what was done in the old code so I'm leaving it as is.
+	if config.all.entity?.deletingRoles?
+		rolesToTest = []
+		for role in config.all.entity.deletingRoles.split(",")
+			role = role.trim()
+			if role is 'entityScientist'
+				if (user.username is scientist)
+					return true
+			else if role is 'projectAdmin'
+				projectAdminRole =
+					lsType: "Project"
+					lsKind: project
+					roleName: "Administrator"
+				if testUserHasRoleTypeKindName(user, [projectAdminRole])
+					return true
+			else
+				rolesToTest.push role
+		if rolesToTest.length is 0
+			return false
+		unless testUserHasRole user, rolesToTest
+			return false
+	return true
+
+exports.getEntityValue = (entity, stateType, stateKind, valueType, valueKind) ->
```

Comment:
This feels more like a global utility, no?

Reply:
Yeah, but waiting to move it there until it gets reused.